src/iop/hlreconstruct/segbased.c

```diff
@@ -198,17 +198,17 @@ static inline float _calc_refavg(const float *in,
   dt_aligned_pixel_t mean = { 0.0f, 0.0f, 0.0f, 0.0f };
   dt_aligned_pixel_t cnt = { 0.0f, 0.0f, 0.0f, 0.0f };
 
-  const int dymin = (row > 0) ? -1 : 0;
-  const int dxmin = (col > 0) ? -1 : 0;
-  const int dymax = (row < roi->height -1) ? 2 : 1;
-  const int dxmax = (col < roi->width -1) ? 2 : 1;
+  const int dymin = MAX(0, row - 1);
+  const int dxmin = MAX(0, col - 1);
+  const int dymax = MIN(roi->height - 1, row +2);
+  const int dxmax = MIN(roi->width - 1, col + 2);
```

Comment:
This changes the code from averaging over a 3x3 square centered at (row,col), with restriction to the RoI, to now averaging over the entire RoI.  Is that intended?